torch/_dynamo/eval_frame.py

```diff
@@ -1449,6 +1449,7 @@ def patch():
         # Note: we don't support sparsity, data-dependent control, or tracing through backwards
         excluded_opts = {torch.optim.SparseAdam, torch.optim.RAdam, torch.optim.LBFGS}
         for opt in optimizers:
+            opt.register_load_state_dict_pre_hook = disable(opt.register_load_state_dict_pre_hook)
```

Comment:
With this change, the tests pass, but I'm not sure this is the best way to go about tracing optimizers @mlazos